### PR TITLE
teleport 16.0.1

### PR DESCRIPTION
**Pull request by itsdalmo**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

`brew audit` fails with the following message:
```
teleport
  * line 32, col 3: Formulae in homebrew/core should use 'depends_on "rust" => build' instead of 'depends_on "rustup-init" => :build'.
  * line 46, col 12: Formula in homebrew/core should not use `rustup-init` at build-time.
Error: 2 problems in 1 formula detected.
```

But if I swap from `rustup-init` to `rust`, the build fails because `wasm-pack` cannot find the wasm32-unknown-unknown target. As such I stuck with `rustup-init` and added a comment which explains why (I believe) it is required.

### CODE REVIEW

**Comment by SMillerDev** [comment]
```
  # github.com/gravitational/teleport/tool/teleport
  /opt/homebrew/Cellar/go/1.22.4/libexec/pkg/tool/darwin_arm64/link: running clang failed: exit status 1
  ld: library not found for -ld_classic
```

**Comment by SMillerDev** [comment]
Thanks for the PR, I agree we're stuck with `rustup` here. I've kicked off another CI run to see if it works on other macOS versions.

**Comment by itsdalmo** [comment]
Sorry, I force pushed a change (before I saw that you had commented and kicked off another CI run) in an attempt to fix the `-ld_classic` error you pasted above.  It seems like the flag was added to the teleport builds 2 weeks ago and backported to [version 16](https://github.com/gravitational/teleport/pull/42738) (as well as 14 and 15). 

Here is the change:
```make
ifeq ("$(OS)-$(ARCH)","darwin-arm64")
# Temporary link flags due to changes in Apple's linker
# https://github.com/golang/go/issues/67854
GO_LDFLAGS += -extldflags=-ld_classic
endif
```

So I suppose this makes xcode a build dependency on darwin, but only on arm64? Need to figure out how to represent this in the formula though.

File: Formula/t/teleport.rb [hunk 1]
```diff
@@ -29,10 +29,16 @@ class Teleport < Formula
...
   depends_on "go" => :build
   depends_on "pkg-config" => :build
+  depends_on "rustup-init" => :build # wasm-pack requires rustup to install the wasm32-unknown-unknown target
+  depends_on "wasm-pack" => :build
   depends_on "yarn" => :build
   depends_on "libfido2"
   depends_on "node"
   depends_on "openssl@3"
+  
+  if on_macos do
```
Comment (SMillerDev) [inline Formula/t/teleport.rb:39]:
```suggestion
  on_macos do
```

**Comment by SMillerDev** [comment]
Linux runs out of disk space and I suspect macOS 12 doesn't have a new enough CLang to recognise that flag.

**Comment by itsdalmo** [comment]
Yep, seems like the only solution is to mark macOS 12 as broken and not build it?

**Comment by SMillerDev** [comment]
Could you ask upstream if building with Xcode 14 is still supposed to be supported? Because that would clarify if we should restrict it or patch it.

**Comment by itsdalmo** [comment]
According to the installation instructions, the minimal version of teleport is `v10.15+ (Catalina)`:
https://goteleport.com/docs/installation

As we've noticed, they use a build flag that requires Xcode 15, which in turn requires macOS 13:
https://developer.apple.com/support/xcode

And as far as I can tell, they only test/build on macos 13:
https://github.com/gravitational/teleport/blob/master/.github/workflows/build-macos.yaml#L22

To me this means that xcode 15 is required to build the binaries, but the resulting binaries will run on any macOS version above `v10.15`. Is it possible for us to do the same, and build the binaries on MacOS 13+, and allow hosts without xcode 15 installed to install the bottle?

**Comment by SMillerDev** [comment]
> Is it possible for us to do the same, and build the binaries on MacOS 13+, and allow hosts without xcode 15 installed to only install from a bottle?

No, Homebrew bottles are only build for the OS they are building on. But it would be good to ask upstream what the idea is. They might not know they broke older builds, so if we ask they can patch it.

**Comment by AndiDog** [comment]
For users wanting the latest `tsh`, for example because of `ERROR: error reading from server: EOF`:

```
brew remove teleport # v14.x at the moment, installs outdated tsh binary
brew install tsh # this is at v16.x already
```

**Comment by itsdalmo** [comment]
Seems like the `tsh` formula downloads binaries that are built by Teleport? In that case it is subject to their new license, and might be in violation if one uses it for work in a company with >100 employees.

**Pull request closed without merging**